extensions/vercast/src/pages/lists/menubar-deployments.tsx

```diff
@@ -0,0 +1,162 @@
+import { Icon, Color, MenuBarExtra, open, Keyboard } from "@raycast/api";
+import useVercel from "../../hooks/use-vercel-info";
+import fromNow from "../../utils/time";
+import { Deployment, DeploymentState } from "../../types";
+import { FetchHeaders, getFetchDeploymentsURL } from "../../vercel";
+import { useFetch } from "@raycast/utils";
+import MenuBarTeamAccessory from "../search-projects/team-switch-menubar-accessory";
+
+interface StateStyles extends Record<DeploymentState, { icon: string; color: string }> {
+  undefined: { icon: string; color: string };
+}
+
+const stateStyles: StateStyles = {
+  READY: {
+    icon: Icon.CheckCircle,
+    color: Color.Green,
+  },
+  BUILDING: {
+    icon: Icon.Bolt,
+    color: Color.Yellow,
+  },
+  INITIALIZING: {
+    icon: Icon.BoltDisabled,
+    color: Color.Yellow,
+  },
+  FAILED: {
+    icon: Icon.ExclamationMark,
+    color: Color.Red,
+  },
+  CANCELED: {
+    icon: Icon.MinusCircleFilled,
+    color: Color.Red,
+  },
+  ERROR: {
+    icon: Icon.Warning,
+    color: Color.Red,
+  },
+  QUEUED: {
+    icon: Icon.List,
+    color: Color.Purple,
+  },
+  undefined: {
+    icon: Icon.Dot,
+    color: Color.PrimaryText,
+  },
+};
+
+const seperator = "·";
+const newline = "\n";
+const numShow = 12;
+
+const MenuBarDeployments = ({ projectId }: { projectId?: string }) => {
+  const { user, selectedTeam } = useVercel();
+  const url = getFetchDeploymentsURL(selectedTeam?.id, projectId);
+
+  const { isLoading, data, revalidate } = useFetch<{
+    deployments: Deployment[];
+    // TODO: why can't I `{ headers: FetchHeaders }` here?
+  }>(url, {
+    // @ts-expect-error Type 'null' is not assignable to type 'string'.
+    headers: FetchHeaders.get("Authorization") ? [["Authorization", FetchHeaders.get("Authorization")]] : [[]],
+  });
+
+  const deployments = data?.deployments;
+
+  const onTeamChange = () => {
+    revalidate();
+  };
+
+  const mostRecentDeploymentState = deployments?.[0]?.state || "undefined";
+
+  return (
+    <MenuBarExtra
+      icon={{
+        source: "../assets/vercel.svg",
+        tintColor:
+          mostRecentDeploymentState === "READY" // If the most recent deployment state is READY change the tint to white or dark depending on system theme
+            ? { light: "000000", dark: "FFFFFF" }
+            : stateStyles[mostRecentDeploymentState].color,
```

Comment:
Would it be better to add this behavior to the `stateStyles` object?

What I understand is that, for the "READY" case in the `MenuBarExtra` we want it to be either black or white, but in the `MenuBarExtra.Section` we want it to be green. Is that correct?

I was thinking something like the following, but I'm not sure if it's actually better. Just thinking out loud in case you prefer it:

```ts
const stateStyles: StateStyles = {
  READY: {
    icon: Icon.CheckCircle,
    color: Color.Green,
    themedColor: { light: "000000", dark: "FFFFFF" }
  },
  BUILDING: {
    icon: Icon.Bolt,
    color: Color.Yellow,
  },
  // ...
}
```

Then on the component:
```ts
<MenuBarExtra
  icon={{
    source: "../assets/vercel.svg",
    tintColor:
      stateStyles[mostRecentDeploymentState].themedColor ?? stateStyles[mostRecentDeploymentState].color,
  }}
  isLoading={isLoading}
>
...
```

That way, if the `themedColor` property is defined for the `mostRecentDeploymentState`, it uses it, and it defaults to the `color` property if it isn't defined.

I find it a bit cleaner as it removes the exceptional case for "READY" and makes it more generic, but you might prefer the ternary because it's more explicit. I'm happy with both, but just wanted to share this option.